src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509ChainPolicy.cs

```diff
@@ -94,10 +95,32 @@ public X509ChainTrustMode TrustMode
             }
         }
 
-        public DateTime VerificationTime { get; set; }
+        public DateTime VerificationTime
+        {
+            get
+            {
+                return _verificationTime.Equals(default(DateTime)) ? DateTime.Now : _verificationTime;
+            }
+            set
+            {
+                _verificationTime = value;
+            }
+        }
 
         public TimeSpan UrlRetrievalTimeout { get; set; }
 
+        public X509ChainPolicy Duplicate()
+        {
+            var dup = (X509ChainPolicy)this.MemberwiseClone();
```

Comment:
Everything should just be cloned manually.  All of the collection values need to be cloned to avoid delayed or concurrent modification (e.g. you're potentially adding to ApplicationPolicy, which could be an empty, non-null collection).

Reply:
could be done. I did not want to duplicate things we don't really need to save allocations.

Reply:
I was thinking about it. I had offline chat with @bartonjs and this PR is out to flesh out possible behavior to API  proposed in https://github.com/dotnet/runtime/issues/59979#issuecomment-1124394748 . The policy is not in center but we should make sure we can achieve reasonable behavior. (and solve the underlying use cases (listed only one here))